src/Microsoft.Azure.WebJobs.Host/Bindings/PatternMatcher.cs

```diff
@@ -268,21 +329,22 @@ private static bool CheckArg(Type openType, Type specificType, Dictionary<string
         // Helper for getting a converter func that invokes the converter interface on an object.
         private abstract class InvokerBase
         {
-            public abstract Func<object, object> Work(object instance);
+            // This should get changed to a FuncAsyncConverter 
+            public abstract FuncAsyncConverter Work(object instance);
         }
 
         // Get a converter function that invokes IConverter on an object. 
         private class Invoker<TSrc, TDest> : InvokerBase
         {
-            public override Func<object, object> Work(object instance)
+            public override FuncAsyncConverter Work(object instance)
```

Comment:
The "Work" naming is strange. I'd expect InvokerBase to have an Invoke, or Execute method.